tests/conftest.py

```diff
@@ -9,15 +9,28 @@
 
 import bionic as bn
 
+PARALLEL_PROCESSING_TYPE = "PARALLEL"
+SERIAL_PROCESSING_TYPE = "SERIAL"
 
-@pytest.fixture(scope="session")
-def parallel_processing_enabled(request):
-    return request.config.getoption("--parallel")
+
+# Parameterizing a fixture adds the parameter in the test name at the end,
+# like test_name[PARALLEL] and test_name[SERIAL]. This is super helpful while
+# debugging and much clearer than parameterizing `parallel_processing_enabled`
+# which suffixes [TRUE] / [FALSE].
+@pytest.fixture(params=[PARALLEL_PROCESSING_TYPE, SERIAL_PROCESSING_TYPE])
+def processing_type(request):
+    return request.param
+
+
+@pytest.fixture
+def parallel_processing_enabled(processing_type):
+    global PARALLEL_PROCESSING_TYPE
```

Comment:
Okay, this one is because I haven't worked much with global variables in python. The only time I used global variables before was to cache objects (and hence write to global variables) which requires the use of `global`. After your comment, I read up more on it and it doesn't seem like I need to use `global` when just reading the global variable. Going to remove it, thanks!